### DISCUSSION TITLE
High Image Generation Time When Loading Specific LoRAs

### DISCUSSION THREAD

**Discussion by BellZwei** [lllyasviel/stable-diffusion-webui-forge#654] [Ideas]
I'm encountering significantly increased image generation times (upwards of 5 minutes) when loading certain LoRAs for the first time, then after few generations, the speed is normal again (around 15 seconds).

Any Ideas on how to resolve this issue.

Hardware:
CPU: [Intel Core i7-8700K]
GPU: [NVIDIA GeForce RTX 4070Ti]
VRAM: [12GB]
RAM: [32GB]

**Comment by BellZwei** [comment]
Am I the only one who has this problem?

**Reply by AI-Ilikeit** [reply]
i get this too, i'll cancel the gen, and then it's back to normal until i change a new lora again.

**Reply by c4dee** [reply]
Nope, same problem here. I switched back to A1111.

**Comment by willhsmit** [comment]
Same here (RTX 3060 12 GB); for a 768x1024 it drops from 1.8 it/s to 13 s/it before returning to normal on the next generation (even if the next generation is the same lora and prompt).

One pattern I notice is that the Model Memory shows 0.0 only for these slow generations. Other than that there's no difference to the output from the commandline.

[Memory Management] Current Free GPU Memory (MB) =  3601.97705078125
[Memory Management] Model Memory (MB) =  0.0
[Memory Management] Minimal Inference Memory (MB) =  1024.0
[Memory Management] Estimated Remaining GPU Memory (MB) =  2577.97705078125

When I check Task Manager (this is Windows), the CPU usage is low and the GPU usage high, so it is using the GPU to compute, but maybe it is reading the model or writing the output from main memory?

**Comment by ostap667inbox** [comment]
Turn on the resource manager when generating. This is a known bug of Forge - sometimes VRAM leaks when LoRa is connected. 
As soon as you see that generation has suddenly taken up 100% of VRAM and started using slow system memory, interrupt generation and perform some action that will force VRAM management. Personally, I turn on 'Never OOM' mode, start and immediately abort the generation. This is fast enough and guaranteed to clear the VRAM.
